metrics/provider/providers.go

```diff
@@ -0,0 +1,237 @@
+}
+
+// NewGauge implements Provider. Help is ignored.
+func (p statsdProvider) NewGauge(name, _ string) metrics.Gauge {
+	return p.e.NewGauge(name)
+}
+
+// Stop will call the underlying statsd.Emitter's Stop method.
+func (p statsdProvider) Stop() {
+	p.e.Stop()
+}
+
+// NewDogStatsdProvider will return a Provider implementation that is a simple
+// wrapper around a dogstatsd.Emitter. All metric names will be prefixed with
+// the given value and data will be emitted once every interval or when the
+// buffer has reached its max size. If no network value is given, it will
+// default to "udp".
+func NewDogStatsdProvider(network, address, prefix string, interval time.Duration, logger log.Logger) (Provider, error) {
+	if network == "" {
+		network = "udp"
+	}
+	if address == "" {
+		return nil, errors.New("address is required")
+	}
+	return dogstatsdProvider{
+		e: dogstatsd.NewEmitter(network, address, prefix, interval, logger),
+	}, nil
+}
+
+type dogstatsdProvider struct {
+	e *dogstatsd.Emitter
+}
+
+var _ Provider = dogstatsdProvider{}
+
+// NewCounter implements Provider. Help is ignored.
+func (p dogstatsdProvider) NewCounter(name, _ string) metrics.Counter {
+	return p.e.NewCounter(name)
+}
+
+// NewHistogram implements Provider. Help is ignored.
+func (p dogstatsdProvider) NewHistogram(name, _ string, min, max int64, sigfigs int, quantiles ...int) (metrics.Histogram, error) {
+	return p.e.NewHistogram(name), nil
+}
+
+// NewGauge implements Provider. Help is ignored.
+func (p dogstatsdProvider) NewGauge(name, _ string) metrics.Gauge {
+	return p.e.NewGauge(name)
+}
+
+// Stop will call the underlying statsd.Emitter's Stop method.
+func (p dogstatsdProvider) Stop() {
+	p.e.Stop()
+}
+
+// NewExpvarProvider is a very thin wrapper over the expvar package.
+// If a prefix is provided, it will prefix all metric names.
+func NewExpvarProvider(prefix string) Provider {
+	return expvarProvider{prefix: prefix}
+}
+
+type expvarProvider struct {
+	prefix string
+}
+
+var _ Provider = expvarProvider{}
+
+// NewCounter implements Provider. Help is ignored.
+func (p expvarProvider) NewCounter(name, _ string) metrics.Counter {
+	return kitexp.NewCounter(p.prefix + name)
+}
+
+// NewHistogram implements Provider. Help is ignored.
+func (p expvarProvider) NewHistogram(name, _ string, min, max int64, sigfigs int, quantiles ...int) (metrics.Histogram, error) {
+	return kitexp.NewHistogram(p.prefix+name, min, max, sigfigs, quantiles...), nil
+}
+
+// NewGauge implements Provider. Help is ignored.
+func (p expvarProvider) NewGauge(name, _ string) metrics.Gauge {
+	return kitexp.NewGauge(p.prefix + name)
+}
+
+// Stop is a no-op.
+func (expvarProvider) Stop() {}
+
+type prometheusProvider struct {
+	namespace string
+	subsystem string
+}
+
+var _ Provider = prometheusProvider{}
+
+// NewPrometheusProvider returns a Prometheus provider that uses the provided
+// namespace and subsystem for all metrics. Help is not provided, which is very
```

Comment:
Do we still need this note since we've added the "help" parameter to all of the implementations?